### ISSUE TITLE
npm run generate -- component Throwing error

### ISSUE THREAD

**Issue by HaakonH** [ngx-rocket/generator-ngx-rocket#140]
## I'm submitting a...

[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request

## Current behavior

After successfully building and running ngx-rocket generator and running "npm run generate -- component someComponentName"  getting an error: "Schematic "someComponentName" not found in collection "@schematics/angular". "

## Expected behavior

Expecting that a new component would be generated.

## Minimal reproduction of the problem with instructions

1. Open a command window in a folder where you have successfully build and run a ngx-rocket generator.
2. Run the command "npm run generate --component someComponentName"
3. Error will be displayed - "Schematic "someComponentName" not found in collection "@schematics/angular". "

## Environment

```
- generator version: 1.0.3
- node version: v6.9.1
- npm version: 3.10.8 
- OS: Windows
```

Others:

**Comment by sinedied** [comment]
Seems like you have an incorrect install of angular-cli, it needs special care when updating, see: https://github.com/angular/angular-cli#updating-angular-cli to make sure you have a clean local install (global install is not needed here)

**Comment by HaakonH** [comment]
Hi sinedied,

It turns out that the problem is one of me being literal with the script instructions and the instructions being wrong.

I tried to run this code in the cmd window after following your instructions about updating the angular cli.

**npm run generate --component coding**

Turns out that the script to create component looks like this instead:

**npm run generate component coding**

So this looks like an easy fix, just remove the "--" from before the component declaration. I'm one of these coders that, you know, takes things as literal instructions and follow them to the letter because that is how things work. 

I'd personally remove that and clarify the documentation if I could so that it helps other devs who want to use this great Yeoman package.

Thanks again for your help with this issue!

**Comment by sinedied** [comment]
@HaakonH Ah, I missed that one, it was a simple space issue: there is a space between `--` and `component`, that was your issue (the space is in the docs, but it's easy to miss).

The `--` alone tells npm to pass-through arguments and flags to the underlying command, if you remove it it will work for text arguments but not for flags (for example `--module <name>` so it's better to use it all the time.

To avoid those kind of issues, I just released a new `1.4.0` CLI version that removed the need for `--` at all if you use it, ie you can simply type:
`ngx generate component someComponent --module someModule`
